diff --git a/src/librabbitmq/amqp_api.c b/src/librabbitmq/amqp_api.c
index 033f84a..f8e87ed 100644
--- a/src/librabbitmq/amqp_api.c
+++ b/src/librabbitmq/amqp_api.c
@@ -3,6 +3,7 @@
 #include <string.h>
 #include <stdint.h>
 #include <errno.h>
+#include <sys/select.h>
 
 #ifndef DISABLE_THREADS
 #include <pthread.h>
diff --git a/src/librabbitmq/amqp_connection.c b/src/librabbitmq/amqp_connection.c
index a1ec1cd..69e91e5 100644
--- a/src/librabbitmq/amqp_connection.c
+++ b/src/librabbitmq/amqp_connection.c
@@ -3,6 +3,7 @@
 #include <string.h>
 #include <stdint.h>
 #include <errno.h>
+#include <sys/select.h>
 
 #include <unistd.h>
 #include <sys/uio.h>
diff --git a/src/librabbitmq/amqp_framing.c b/src/librabbitmq/amqp_framing.c
index 258bd66..11aae16 100644
--- a/src/librabbitmq/amqp_framing.c
+++ b/src/librabbitmq/amqp_framing.c
@@ -3,6 +3,7 @@
 #include <stdio.h>
 #include <errno.h>
 #include <arpa/inet.h> /* ntohl, htonl, ntohs, htons */
+#include <sys/select.h>
 
 #include "amqp.h"
 #include "amqp_framing.h"
diff --git a/src/librabbitmq/amqp_mem.c b/src/librabbitmq/amqp_mem.c
index da7c0d6..296a861 100644
--- a/src/librabbitmq/amqp_mem.c
+++ b/src/librabbitmq/amqp_mem.c
@@ -5,6 +5,7 @@
 #include <sys/types.h>
 #include <errno.h>
 #include <assert.h>
+#include <sys/select.h>
 
 #include "amqp.h"
 #include "amqp_config.h"
diff --git a/src/librabbitmq/amqp_socket.c b/src/librabbitmq/amqp_socket.c
index a9aa8fc..81b79c8 100644
--- a/src/librabbitmq/amqp_socket.c
+++ b/src/librabbitmq/amqp_socket.c
@@ -6,6 +6,7 @@
 #include <stdarg.h>
 #include <fcntl.h>
 #include <poll.h>
+#include <sys/select.h>
 
 #include "amqp.h"
 #include "amqp_framing.h"
diff --git a/src/librabbitmq/amqp_table.c b/src/librabbitmq/amqp_table.c
index 16d85ff..1b8770a 100644
--- a/src/librabbitmq/amqp_table.c
+++ b/src/librabbitmq/amqp_table.c
@@ -3,6 +3,7 @@
 #include <string.h>
 #include <stdint.h>
 #include <errno.h>
+#include <sys/select.h>
 
 #include "amqp.h"
 #include "amqp_private.h"
diff --git a/src/pg_amqp.c b/src/pg_amqp.c
index 1e091ac..0572d14 100644
--- a/src/pg_amqp.c
+++ b/src/pg_amqp.c
@@ -37,6 +37,7 @@
 #include <time.h>
 #include <sys/time.h>
 #include <sys/socket.h>
+#include <sys/select.h>
 
 #include "postgres.h"
 #include "funcapi.h"
